backend/de.metas.material/planning/src/main/java/de/metas/material/planning/event/SupplyRequiredDecreasedNotificationProducer.java

```diff
@@ -0,0 +1,139 @@
+			.type(Type.DISTRIBUTED)
+			.build();
+	@NonNull private final INotificationBL notificationBL = Services.get(INotificationBL.class);
+	@NonNull private final IOrgDAO orgDAO = Services.get(IOrgDAO.class);
+	@NonNull private final IOrderBL orderBL = Services.get(IOrderBL.class);
+
+	public static SupplyRequiredDecreasedNotificationProducer newInstance()
+	{
+		return new SupplyRequiredDecreasedNotificationProducer();
+	}
+
+	private SupplyRequiredDecreasedNotificationProducer()
+	{
+	}
+
+	public void sendNotification(@Nullable final MaterialPlanningContext context, @NonNull final SupplyRequiredDescriptor descriptor, @NonNull final Quantity unhandledQty)
+	{
+		final TableRecordReference recordReference = getTableRecordReferenceFromEventDescriptor(descriptor);
+		if (recordReference == null)
+		{
+			return;
+		}
+
+		final UserNotificationRequest.UserNotificationRequestBuilder notificationBuilder = UserNotificationRequest.builder()
+				.topic(EVENTBUS_TOPIC)
+				.targetAction(UserNotificationRequest.TargetRecordAction.ofNullable(recordReference))
+				.contentADMessage(MSG_SUPPLY_REQUIRED_DECREASED)
+				.contentADMessageParam(unhandledQty.toString());
+
+		getRecipients(context, descriptor).forEach(userId -> {
+			notificationBuilder.recipientUserId(userId);
+			notificationBL.sendAfterCommit(notificationBuilder
+					.build());
+		});
+	}
+
+	@Nullable
+	private TableRecordReference getTableRecordReferenceFromEventDescriptor(final SupplyRequiredDescriptor descriptor)
+	{
+		final TableRecordReference recordReference = TableRecordReference.ofNullable(I_C_Order.Table_Name, descriptor.getOrderId());
+		if (recordReference == null)
+		{
+			Loggables.withLogger(log, Level.ERROR).addLog("Could not trigger notification for event descriptor {}. Could not identify a DB record to reference.", descriptor);
+			return null;
+		}
+		return recordReference;
+	}
+
+	private Set<UserId> getRecipients(@Nullable final MaterialPlanningContext context, @NonNull final SupplyRequiredDescriptor descriptor)
+	{
+		final Set<UserId> userIds = new HashSet<>();
+		final UserId plannerId = context == null ? null : context.getProductPlanning().getPlannerId();
+		if (plannerId != null && plannerId.isRegularUser())
+		{
+			userIds.add(plannerId);
+		}
+		final OrderId orderId = OrderId.ofRepoIdOrNull(descriptor.getOrderId());
+		if (orderId != null)
+		{
+			final I_C_Order order = orderBL.getById(orderId);
+			final UserId salesRepUserId = UserId.ofRepoIdOrNull(order.getSalesRep_ID());
+			if (salesRepUserId != null && salesRepUserId.isRegularUser())
+			{
+				userIds.add(salesRepUserId);
+			}
+
+			final UserId updatedByUserId = UserId.ofRepoIdOrNull(InterfaceWrapperHelper.getPO(order).getUpdatedBy());
+			if (updatedByUserId != null && updatedByUserId.isRegularUser())
+			{
+				userIds.add(updatedByUserId);
+			}
+		}
+
+		//deadletter recipient
+		final UserId supervisorId = orgDAO.getOrgInfoById(descriptor.getOrgId()).getSupervisorId();
```

Comment:
pls get the supervisorId only if userIds is empty

Reply:
done